odoo/tools/translate.py

```diff
@@ -1255,5 +1343,94 @@ def load_language(cr, lang):
         l10n flavor (ex: 'fr', 'fr_BE', but not 'fr-BE')
     """
     env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
-    installer = env['base.language.install'].create({'lang_ids': [(6, 0, [env['res.lang'].search([('code', '=', lang)]).id])]})
+    lang_ids = env['res.lang'].with_context(active_test=False).search([('code', '=', lang)]).ids
+    installer = env['base.language.install'].create({'lang_ids': [(6, 0, lang_ids)]})
     installer.lang_install()
+
+
+class CodeTranslations:
+    def __init__(self):
+        # {(module_name, lang): {src: value}}
+        self.python_translations = {}
+        # {(module_name, lang): {'message': [{'id': src, 'string': value}]}
+        self.web_translations = {}
+
+    def _get_po_paths(self, mod, lang):
+        lang_base = lang.split('_')[0]
+        po_paths = [get_resource_path(mod, 'i18n', lang_base + '.po'),
+                    get_resource_path(mod, 'i18n', lang + '.po'),
+                    get_resource_path(mod, 'i18n_extra', lang_base + '.po'),
+                    get_resource_path(mod, 'i18n_extra', lang + '.po')]
+        return [path for path in po_paths if path]
+
+    def _trans_load_code_python(self, fileobj, fileformat, lang):
+        try:
+            python_translations = {}
+            fileobj.seek(0)
+            reader = TranslationFileReader(fileobj, fileformat=fileformat)
+            for row in reader:
+                if row.get('value') and row.get('src') and row.get('type') == 'code':
+                    python_translations[row['src']] = row['value']
+            return python_translations
+
+        except IOError:
+            iso_lang = get_iso_codes(lang)
+            filename = '[lang: %s][format: %s]' % (iso_lang or 'new', fileformat)
+            _logger.exception("couldn't read translation file %s", filename)
```

Comment:
This catches and lots the exception, then doesn't return antyhign. As a result the caller (`_load_python_translations`) will just crash when it tries to update the existing translations set with the new translations.

That seems sub-optimal. Either log and suppress the exception or let it bubble (possibly wrapped), but don't log then trigger a new unrelated error.